cmake/macros/SetupHifiLibrary.cmake

```diff
@@ -59,7 +59,7 @@ macro(SETUP_HIFI_LIBRARY)
   if (${${TARGET_NAME}_SHARED})
     add_library(${TARGET_NAME} SHARED ${LIB_SRCS} ${AUTOSCRIBE_SHADER_LIB_SRC} ${QT_RESOURCES_FILE})
   else ()
-    add_library(${TARGET_NAME} ${LIB_SRCS} ${AUTOSCRIBE_SHADER_LIB_SRC} ${QT_RESOURCES_FILE})
+    add_library(${TARGET_NAME} STATIC ${LIB_SRCS} ${AUTOSCRIBE_SHADER_LIB_SRC} ${QT_RESOURCES_FILE})
```

Comment:
This is probably going to make things crash. We have a problem here which we bumped into before.

We have many libraries that form a dependency tree. Something like this:

* shared linked into A
* shared linked into B
* libA and libB linked into binary

The problem is that when linking statically, constructors and destructors get duplicated. The final binary gets two copies of the 'shared' code, one from libA, and another from libB.

Then at runtime stuff starts crashing because for instance constructors and destructors get run twice.

Linking everything dynamically fixed that issue because the dynamic linker resolves that problem. There's a past PR in which I changed static linking to dynamic for this exact reason, I will try to dig up a link to it.

Reply:
Found it, change was in PR #158

Reply:
I see, in that case I'll add another option for client library packaging based on the same BUILD_SHARED_LIBS setting instead of this hack.

Reply:
Done. I'll keep this in mind though in context of the client library going forward. No static linking specific crashes there so far, but might be just luck.